api/v1beta2/ibmpowervscluster_types.go

```diff
@@ -243,6 +245,8 @@ type IBMPowerVSClusterList struct {
 // TransitGateway holds the TransitGateway information.
 type TransitGateway struct {
 	// name of resource.
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Pattern=`^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$`
```

Comment:
Did you manually verified from UI this before adding the check?

Reply:
Here is the reference https://cloud.ibm.com/apidocs/transit-gateway#create-transit-gateway, It does support capital letters